src/chat.py

```diff
@@ -318,13 +318,17 @@ async def handle_user_message( # pylint: disable=too-many-locals,too-many-statem
 
         async with cl.Step(name="thinking and generating a response") as resp_step:
             # Process user message and get AI response
+            if not search_results:
+                temperature = config.default_temperature_without_search_results
+            else:
+                temperature= settings["temperature"]
```

Comment:
Small nit:). Not important.
```suggestion
                temperature = settings["temperature"]
```